### PR TITLE
Refactoring server

### PR DESCRIPTION
**Pull request by katestam**

### CODE REVIEW

**Review by dwrz — APPROVED** [review]

File: server/db/controllers/articleController.js [hunk 1]
```diff
@@ -34,20 +35,22 @@ module.exports = {
           cb(article);
         })
         .catch((err) => {
-          cb('Error saving article to database:', err);
+          throw ('Error saving article to database:', err);
```
Comment (therobinkim) [inline server/db/controllers/articleController.js:38]: did you mean to throw an error here? (is it being caught and handled elsewhere?)

File: server/server-index.js [hunk 2]
```diff
@@ -1,113 +1,112 @@
...
+//   }
+// });
+
+// app.get('/api/user/allArticles', (req, res) => {
+//   userDataGetter.getArticlesByUser(req.session.user, (articles) => {
+//     res.send(articles);
+//   });
+// });
+
+<<<<<<< HEAD
```
Comment (therobinkim) [inline server/server-index.js:109]: this looks like a leftover artifact from a merge conflict

File: server/server-index.js [hunk 3]
```diff
@@ -116,3 +115,8 @@ app.listen(PORT, () => {
   console.log(`Listening on ${PORT}`);
 });
 
+=======
```
Comment (therobinkim) [inline server/server-index.js:118]: this looks like a leftover artifact from a merge conflict

File: server/server-index.js [hunk 4]
```diff
@@ -116,3 +115,8 @@ app.listen(PORT, () => {
...
+// app.listen(PORT, () => {
+//   console.log(`Listening on ${PORT}`);
+// });
+>>>>>>> Fix promise pyramid
```
Comment (therobinkim) [inline server/server-index.js:122]: this looks like a leftover artifact from a merge conflict

**Pull request merged**